aapcs64-morello/aapcs64-morello.rst

```diff
@@ -0,0 +1,858 @@
+    | r30       | LR       | The Link Register.                                                                                 |
+    +-----------+----------+----------------------------------------------------------------------------------------------------+
+    | r29       | FP       | The Frame Pointer.                                                                                 |
+    +-----------+----------+----------------------------------------------------------------------------------------------------+
+    | r19-r28   |          | The lower 64 bits of the registers (x19-x28) is callee-saved.                                      |
+    +-----------+----------+----------------------------------------------------------------------------------------------------+
+    | r18       |          | The Platform Register, if needed; otherwise a temporary register. See notes.                       |
+    +-----------+----------+----------------------------------------------------------------------------------------------------+
+    | r17       | IP1      | The second intra-procedure-call temporary register (can be used by call veneers and PLT code).     |
+    +-----------+----------+----------------------------------------------------------------------------------------------------+
+    | r16       | IP0      | The first intra-procedure-call scratch register (can be used by call veneers and PLT code).        |
+    +-----------+----------+----------------------------------------------------------------------------------------------------+
+    | r9-r15    |          | Temporary registers.                                                                               |
+    +-----------+----------+----------------------------------------------------------------------------------------------------+
+    | r8        |          | The indirect result location register.                                                             |
+    +-----------+----------+----------------------------------------------------------------------------------------------------+
+    | r0-r7     |          | Parameter/result registers.                                                                        |
+    +-----------+----------+----------------------------------------------------------------------------------------------------+
+
+
+The first eight registers, r0-r7, are used to pass argument values into a subroutine and to return result values from a function. They may also be used to hold intermediate values within a routine (but, in general, only between subroutine calls).
+
+Registers r16 (IP0/CIP0) and r17 (IP1/CIP1) may be used by a linker as a scratch register between a routine and any subroutine it calls (for details, see `Capabilities`_). They can also be used within a routine to hold intermediate values between subroutine calls.
```

Comment:
It looks like the `Cababilities`_ link here is wrong. Perhaps done through a script?

Reply:
I did it manually. Looks correct to me because it goes to the 5.2 Capabilities section both in my local pdf and github rendering as expected.

Reply:
But in the pdf of the original release up on developer, this isn't a link and reads: (for details, see aapcs64-section5-3-1)

Which I think must refer to section 6.3.1 of the aapcs64 spec up on Github: https://github.com/ARM-software/abi-aa/blob/master/aapcs64/aapcs64.rst#id55 (I wouldn't link to that though, I think), which actually deals with IP0/IP1, as opposed to the Capabilities section in this document.

Reply:
Ah yes, sorry I was looking at the table in section 5.1 which also links correctly to Capabilities.

The link you point out seems to be a bug in the original spec which linked to the non-existent aapcs64-section5-3-1 section, when it should have been using _aapcs64-morello-section5-3-1, which is for the section "Use of CIP0 and CIP1 by the linker". Will correct it.